applications/StructuralMechanicsApplication/tests/cpp_tests/test_high_cycle_fatigue_constitutive_law_3d.cpp

```diff
@@ -0,0 +1,155 @@
+    strain_vector[3] = 0.0;
+    strain_vector[4] = 0.0;
+    strain_vector[5] = 0.0;
+
+    material_properties.SetValue(DENSITY, 7850.0);
+    material_properties.SetValue(YOUNG_MODULUS, 2.01e11);
+    material_properties.SetValue(POISSON_RATIO, 0.3);
+    material_properties.SetValue(FRACTURE_ENERGY, 20e5);
+    material_properties.SetValue(YIELD_STRESS, 838.9e6);
+    material_properties.SetValue(SOFTENING_TYPE, 1);
+    Vector high_cycle_fatigue_parameters(7);
+    high_cycle_fatigue_parameters[0]=0.5;
+    high_cycle_fatigue_parameters[1]=0.7;
+    high_cycle_fatigue_parameters[2]=0.5;
+    high_cycle_fatigue_parameters[3]=0.0068;
+    high_cycle_fatigue_parameters[4]=3.35;
+    high_cycle_fatigue_parameters[5]=0.0133;
+    high_cycle_fatigue_parameters[6]=0.0068;
+    material_properties.SetValue(HIGH_CYCLE_FATIGUE_PARAMETERS, high_cycle_fatigue_parameters);
+
+    // Set constitutive law flags:
+    Flags& ConstitutiveLawOptions = cl_parameters.GetOptions();
+    ConstitutiveLawOptions.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, true);
+    ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_STRESS, true);
+    ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false);
+
+    cl_parameters.SetElementGeometry(Geom);
+    cl_parameters.SetProcessInfo(test_model_part.GetProcessInfo());
+    cl_parameters.SetMaterialProperties(material_properties);
+    cl_parameters.SetStrainVector(strain_vector);
+    cl_parameters.SetStressVector(stress_vector);
+    Matrix const_matrix;
+    cl_parameters.SetConstitutiveMatrix(const_matrix);
+
+    // Create the CL's
+    GenericSmallStrainHighCycleFatigueLaw<GenericConstitutiveLawIntegratorDamage<VonMisesYieldSurface<ModifiedMohrCoulombPlasticPotential<6>>>> HCF_CL = GenericSmallStrainHighCycleFatigueLaw<GenericConstitutiveLawIntegratorDamage<VonMisesYieldSurface<ModifiedMohrCoulombPlasticPotential<6>>>>();
+
+    HCF_CL.SetFatigueReductionFactor(8.8914e-01);
+    Vector previous_stresses=ZeroVector(2);
+    previous_stresses[0]=7.6380e+08;
+    previous_stresses[1]=6.9760e+08;
+    
+    HCF_CL.SetPreviousStresses(previous_stresses);
+    HCF_CL.SetMaxStress(0.0);
+    HCF_CL.SetMinStress(0.0);
+    HCF_CL.SetPreviousMaxStress(7.6380e+08);
+    HCF_CL.SetPreviousMinStress(-2.2914e+08);
+    HCF_CL.SetNumberOfCycles(984);
+    HCF_CL.SetReversionFactor(-0.3);
+    HCF_CL.SetCycleCounter(false);
+    HCF_CL.SetFatigueReductionParameter(5.3239E-07);
+
+
+
+    std::vector<double> expected_value;
+    expected_value = {1.07636e+09,4.61299e+08,4.61299e+08,0,0,0};
+
+    double TestDamage;
+    Vector TestStress;
+    Vector dummy;
+    HCF_CL.InitializeMaterial(material_properties, Geom, dummy);
+    HCF_CL.CalculateMaterialResponseCauchy(cl_parameters);
+    HCF_CL.FinalizeMaterialResponseCauchy(cl_parameters);
+    HCF_CL.GetValue(DAMAGE, TestDamage);
+    KRATOS_WARNING_IF("TestHCF", TestDamage < 1.0e-12) << "VonMises:: This test is not in damage range" << std::endl;
```

Comment:
why don't you explicitly test this value?
This is a test, it should be deterministic

Reply:
This was just a check to ensure that the test was working in non-linear domain but I have removed this part now. Thanks.